### PR TITLE
OCD-3228: Add job to remove the 2015 Upload Template (v17)

### PR DESCRIPTION
**Pull request by kekey1**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public void execute(final JobExecutionContext jobContext) throws JobExecutionException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        LOGGER.info("********* Starting the Remove Upload Template v17 job. *********");
+        List<UploadTemplateVersionDTO> uploadTemplates = uploadTemplateDao.findAll();
+        boolean foundTemplate = false;
+        for (UploadTemplateVersionDTO uploadTemplate : uploadTemplates) {
+            if (uploadTemplate.getName().equals(TEMPLATE_NAME)) {
+                foundTemplate = true;
+                try {
+                    uploadTemplateDao.delete(uploadTemplate.getId());
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java]: Does this work without a transaction? I had to wrap mine in one to get it to add a test tool

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ */
+public class RemoveV17UploadTemplate extends QuartzJob {
+    private static final Logger LOGGER = LogManager.getLogger("removeV17UploadTemplateJobLogger");
+    private static final String TEMPLATE_NAME = "2015 CHPL Upload Template v12";
+
+    @Autowired
+    private UploadTemplateVersionDAO uploadTemplateDao;
+
+    @Autowired
+    private FF4j ff4j;
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java]: I don't think this is used

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java (hunk 1 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java]: I saw that in yours. This does work, I ran it twice. I don't know why.

Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RemoveV17UploadTemplate.java]: Oh I do know why. The delete method in the DAO is already marked transactional. As to why that is the case.... I don't know!

**Pull request merged**